osu.Game/Overlays/NotificationOverlay.cs

```diff
@@ -205,7 +206,7 @@ private void addPermanently(Notification notification)
             var ourType = notification.GetType();
             int depth = notification.DisplayOnTop ? -runningDepth : runningDepth;
 
-            var section = sections.Children.First(s => s.AcceptedNotificationTypes.Any(accept => accept.IsAssignableFrom(ourType)));
+            var section = sections.Children.Last(s => s.AcceptedNotificationTypes.Any(accept => accept.IsAssignableFrom(ourType)));
```

Comment:
This is a little dodgy... I'd say that something like this is better:

```diff
diff --git a/osu.Game/Overlays/NotificationOverlay.cs b/osu.Game/Overlays/NotificationOverlay.cs
index dd4924d07a..81233b4343 100644
--- a/osu.Game/Overlays/NotificationOverlay.cs
+++ b/osu.Game/Overlays/NotificationOverlay.cs
@@ -114,7 +114,7 @@ private void load()
                                     Children = new[]
                                     {
                                         // The main section adds as a catch-all for notifications which don't group into other sections.
-                                        new NotificationSection(AccountsStrings.NotificationsTitle, new[] { typeof(Notification) }),
+                                        new NotificationSection(AccountsStrings.NotificationsTitle),
                                         new NotificationSection(NotificationsStrings.RunningTasks, new[] { typeof(ProgressNotification) }),
                                     }
                                 }
@@ -206,7 +206,8 @@ private void addPermanently(Notification notification)
             var ourType = notification.GetType();
             int depth = notification.DisplayOnTop ? -runningDepth : runningDepth;
 
-            var section = sections.Children.Last(s => s.AcceptedNotificationTypes.Any(accept => accept.IsAssignableFrom(ourType)));
+            var section = sections.Children.FirstOrDefault(s => s.AcceptedNotificationTypes?.Any(accept => accept.IsAssignableFrom(ourType)) == true)
+                          ?? sections.First();
 
             section.Add(notification, depth);
 
diff --git a/osu.Game/Overlays/Notifications/NotificationSection.cs b/osu.Game/Overlays/Notifications/NotificationSection.cs
index 10c2900d63..895f13ee9c 100644
--- a/osu.Game/Overlays/Notifications/NotificationSection.cs
+++ b/osu.Game/Overlays/Notifications/NotificationSection.cs
@@ -37,13 +37,17 @@ public void Add(Notification notification, float position)
             notifications.Insert((int)position, notification);
         }
 
-        public IEnumerable<Type> AcceptedNotificationTypes { get; }
+        /// <summary>
+        /// Enumerable of notification types accepted in this section.
+        /// If <see langword="null"/>, the section accepts any and all notifications.
+        /// </summary>
+        public IEnumerable<Type>? AcceptedNotificationTypes { get; }
 
         private readonly LocalisableString titleText;
 
-        public NotificationSection(LocalisableString title, IEnumerable<Type> acceptedNotificationTypes)
+        public NotificationSection(LocalisableString title, IEnumerable<Type>? acceptedNotificationTypes = null)
         {
-            AcceptedNotificationTypes = acceptedNotificationTypes.ToArray();
+            AcceptedNotificationTypes = acceptedNotificationTypes?.ToArray();
 
             titleText = title;
         }

```

Reply:
Sure, was just keeping it simple in terms of LoC change mostly.